fairseq/models/transformer/transformer_encoder.py

```diff
@@ -207,8 +207,7 @@ def forward_scriptable(
         x, encoder_embedding = self.forward_embedding(src_tokens, token_embeddings)
 
         # account for padding while computing the representation
-        if has_pads:
-            x = x * (1 - encoder_padding_mask.unsqueeze(-1).type_as(x))
+        x = x * (1 - encoder_padding_mask.unsqueeze(-1).type_as(x) * has_pads.type_as(x))
```

Comment:
isn't `has_pads ` a boolean? 'bool' object doesn't have 'type_as' attribute!

Reply:
has_pads is a `Tensor` object with a bool scalar, which could be converted by `type_as` method on `Tensor` type. That's my understanding.

Reply:
actually, if the device is "xla" for src_token, it would be a bool which is rare but possible.

Let me update the PR to handle that.

Reply:
@moslehpour updated. mind looking again?

Reply:
Great. This should work! 
Just one last comment, can you confirm if torchscript support type casting with `type_as`? if not our translation pipeline will break

Reply:
@moslehpour sure, I just tried that:
```
@torch.jit.script
def foo(x, y):
    return x.type_as(y)

a = foo(torch.tensor(True), torch.ones(3, 2))
print(a)
```
and seems Torchscript can give us correct result:
```
tensor(1.)
```

Reply:
Perfect. Thanks for checking it.